### PR TITLE
Catch the error from the executor task so that it does not fail the schedule silently

### PR DESCRIPTION
**Pull request by rachelhanhan**
Recently we found a bug that D2 executor in StateUpdater cancelled the update schedule silently. In order to fix this, there are 2 places to fix:

1. Error handling for tasks. Whenever this is an error, we just catch the error instead of let it fail the whole schedule.
2. After some investigation, the quarantine logic can throw a runtime exception when the quarantine latency is longer than the check interval. To reach the parity with degrader strategy, this PR makes sure that the quarantineLatency will not exceed the check interval.

There are other design flaws of quarantine logic in relative load balancer, especially around the dynamic latency calculation, but those are not as urgent as fixing the bug that potentially breaks D2.

### CODE REVIEW

File: d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/QuarantineManager.java [hunk 1]
```diff
@@ -99,7 +100,11 @@ public void updateQuarantineState(PartitionState newPartitionState,
       PartitionState oldPartitionState, long clusterAvgLatency)
   {
     long quarantineLatency = Math.max((long) (clusterAvgLatency * _relativeLatencyLowThresholdFactor),
-        MIN_QUANRANTINE_LATENCY_MS);
+        MIN_QUARANTINE_LATENCY_MS);
+    if (quarantineLatency > MAX_QUARANTINE_LATENCY_MS)
```
Comment (rickzx) [inline d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/QuarantineManager.java:104]: nit: `quarantineLatency = min(MAX_QUARANTINE_LATENCY_MS, quarantineLatency)`?

File: d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java [hunk 2]
```diff
@@ -167,12 +167,18 @@ int getFirstValidPartitionId()
...
+      for (Integer partitionId : _partitionLoadBalancerStateMap.keySet())
+      {
+        PartitionState partitionState = _partitionLoadBalancerStateMap.get(partitionId);
+        updateStateForPartition(partitionState.getTrackerClients(), partitionId, partitionState, partitionState.getClusterGenerationId());
+      }
+    } catch (Exception ex)
     {
-      PartitionState partitionState = _partitionLoadBalancerStateMap.get(partitionId);
-      updateStateForPartition(partitionState.getTrackerClients(), partitionId, partitionState, partitionState.getClusterGenerationId());
+      LOG.error("Failed update the event.", ex);
```
Comment (rickzx) [inline d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java:179]: Maybe `"Failed to update the state for service: " + _serviceName`?

**Review by rickzx — CHANGES_REQUESTED** [review]
LGTM

File: d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/QuarantineManager.java (hunk 1 above)
Comment (rachelhanhan) [reply d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/QuarantineManager.java]: Fixed

File: d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java (hunk 2 above)
Comment (rachelhanhan) [reply d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java]: Fixed

**Review by rickzx — APPROVED** [review]
Thanks for the quick fix!

**Pull request merged**